nvme.c

```diff
@@ -4519,10 +4521,10 @@ static int get_feature_id(struct nvme_dev *dev, struct feat_cfg *cfg,
 		cfg->data_len = 0;
 
 	if (cfg->data_len) {
-		if (posix_memalign(buf, getpagesize(), cfg->data_len)) {
+		size = ((cfg->data_len - 1) / getpagesize() + 1) * getpagesize();
```

Comment:
I think this line deserves a comment. Something like 'rounding up size to page size'.